rovers_parte4_dominio.pddl

```diff
@@ -0,0 +1,206 @@
+              (can_traverse ?r - rover ?x - waypoint ?y - waypoint)
+              ; El rover r puede moverse desde el waypoint x al waypoint y.
+
+              (equipped_for_soil_analysis ?r - rover)
+              ; El rover r esta equipado para analisis de suelo.
+
+              (equipped_for_rock_analysis ?r - rover)
+              ; El rover r esta equipado para analisis de rocas.
+
+              (equipped_for_imaging ?r - rover)
+              ; El rover r esta equipado para tomar imagenes.
+
+              (empty ?s - store)
+              ; El almacen s esta vacio.
+
+              (have_rock_analysis ?r - rover ?w - waypoint)
+              ; El rover r tiene un analisis de rocas del waypoint w.
+
+              (have_soil_analysis ?r - rover ?w - waypoint)
+              ; El rover r tiene un analisis de suelo del waypoint w.
+
+              (full ?s - store)
+              ; El almacen s esta lleno.
+
+              (calibrated ?c - camera ?r - rover)
+              ; La camara c en el rover r esta calibrada.
+
+              (supports ?c - camera ?m - mode)
+              ; La camara c soporta el modo m.
+
+              (available ?r - rover)
+              ; El rover r esta disponible.
+
+              (visible ?w - waypoint ?p - waypoint)
+              ; El waypoint w es visible desde el waypoint p.
+
+              (have_image ?r - rover ?o - objective ?m - mode)
+              ; El rover r tiene una imagen del objetivo o en el modo m.
+
+              (communicated_soil_data ?w - waypoint)
+              ; Los datos de suelo del waypoint w han sido comunicados.
+
+              (communicated_rock_data ?w - waypoint)
+              ; Los datos de rocas del waypoint w han sido comunicados.
+
+              (communicated_image_data ?o - objective ?m - mode)
+              ; Los datos de imagen del objetivo o en el modo m han sido comunicados.
+
+              (at_soil_sample ?w - waypoint)
+              ; Hay una muestra de suelo en el waypoint w.
+
+              (at_rock_sample ?w - waypoint)
+              ; Hay una muestra de roca en el waypoint w.
+
+              (visible_from ?o - objective ?w - waypoint)
+              ; El objetivo o es visible desde el waypoint w.
+
+              (store_of ?s - store ?r - rover)
+              ; El almacen s pertenece al rover r.
+
+              (calibration_target ?i - camera ?o - objective)
+              ; El objetivo o es el objetivo de calibracion de la camara i.
+
+              (on_board ?i - camera ?r - rover)
+              ; La camara i esta a bordo del rover r.
+
+              (channel_free ?l - lander)
+              ; El canal de comunicacion del lander l esta libre.
+
+              (on_board ?b - battery ?r - rover)
+              ; Un rover puede tener bateria a bordo.
+
+              (at_charge ?b - battery ?v - batterylevel)
+              ; Una bateria con carga charge_level.
+       )
+
+       ;Moverse
+       ;aqui deberiamos poner la logica de la bateria?
```

Comment:
Should we use battery level in here? I mean, everytime the rover moves, battery discharges one level?